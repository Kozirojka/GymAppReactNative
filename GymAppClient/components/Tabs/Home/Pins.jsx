    import React from 'react';
    import { View, Text, StyleSheet } from 'react-native';

    const Pins = () => {
        return (
            <View style={styles.container}>
                <Text style={styles.title}>Pins</Text>
                <Text style={styles.description}>This is the Pins component.</Text>
            </View>
        );
    };

    const styles = StyleSheet.create({
        container: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: '#f5f5f5',
        },
        title: {
            fontSize: 24,
            fontWeight: 'bold',
            color: '#333',
            marginBottom: 10,
        },
        description: {
            fontSize: 16,
            color: '#666',
        },
    });

    export default Pins;