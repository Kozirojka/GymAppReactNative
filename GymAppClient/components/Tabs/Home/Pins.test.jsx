import React from 'react';
import { render } from '@testing-library/react-native';
import Pins from './Pins'; 

describe('Pins Component', () => {
  it('renders correctly', () => {
    const { getByText } = render(<Pins />);
    
    const titleElement = getByText('Pins');
    expect(titleElement).toBeTruthy();
    
    const descriptionElement = getByText('This is the Pins component.');
    expect(descriptionElement).toBeTruthy();
  });
  
  it('has the correct styles', () => {
    const { getByText } = render(<Pins />);
    
    const titleElement = getByText('Pins');
    const descriptionElement = getByText('This is the Pins component.');
    
    expect(titleElement.props.style).toEqual(
      expect.objectContaining({
        fontSize: 24,
        fontWeight: 'bold',
        color: '#333',
        marginBottom: 10,
      })
    );
    
    expect(descriptionElement.props.style).toEqual(
      expect.objectContaining({
        fontSize: 16,
        color: '#666',
      })
    );
  });
  
  it('snapshot matches', () => {
    const tree = render(<Pins />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});