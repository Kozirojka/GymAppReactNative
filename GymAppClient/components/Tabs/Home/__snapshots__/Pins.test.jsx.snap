// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pins Component snapshot matches 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Pins
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 16,
      }
    }
  >
    This is the Pins component.
  </Text>
</View>
`;
